src/main/java/io/stargate/sgv2/jsonapi/api/model/command/CommandResult.java

```diff
@@ -86,6 +87,21 @@ public CommandResult(List<Error> errors) {
     this(null, null, errors);
   }
 
+  /**
+   * construct a new CommandResult with warnings added in status map
```

Comment:
This is not a constructor , it is not clear that this method returns a new instance without modifying the original. e.g. add() on a list mutates the list

I could imagine that the issue could be that the status map is immutable.  

two options,: 

1. If we want to be immutable, then rename of withWarning(String) and update comments to make it clear new object is returned. I think it unlikely we will add many so ok to just pass string , not list. 
2. If we want to be mutable, stick with addWarning(string) (again no need for list here). 

In either case, enforce this with a canonical constructor for the record that makes defensive copies and of the status and errors, either immutable or mutable. And same for MultiResponseData as we are here, just List,.copyOf() we are not going to deep clone the JSON node(s) 

While I normally go with immutable objects, in this situation my feeling is make it mutable because results like this can be built up 

as a final nitpick, this may be better organised if at the bottom of the class

Reply:
Thanks, Resolved.